### ISSUE TITLE
Getting Nan in matmul gradient

### ISSUE THREAD

**Issue by saniazahan** [lucidrains/vit-pytorch#84]
![Capture](https://user-images.githubusercontent.com/12523634/112992291-9f632c80-919a-11eb-8f0c-99194cea78c0.PNG)
Hi I am trying to train your ViT model with a different dataset, ( a video of shape [B * 300 * 75] where 300 is the no of frames. So I changed the to_embedding as below to match my data 
```
self.to_embedding = nn.Sequential(
            nn.Linear(self.num_features, dim),
        )
```
unfortunately I am getting nan after some epoch. I have added the backward trace. Could you kindly tell what might be the problem

**Comment by lucidrains** [comment]
if you are working with video, try https://github.com/lucidrains/STAM-pytorch or https://github.com/lucidrains/TimeSformer-pytorch

**Comment by lucidrains** [comment]
https://arxiv.org/abs/2103.15691 further corroborates the conclusions of the two papers above. in fact STAM is just model 2 in this paper, and TimeSformer model 3

**Comment by saniazahan** [comment]
> if you are working with video, try https://github.com/lucidrains/STAM-pytorch or https://github.com/lucidrains/TimeSformer-pytorch

Will give it a try. Thanks
